packages/make-styles/src/makeStyles.ts

```diff
@@ -36,41 +37,55 @@ export function makeStyles<Selectors, Tokens>(definitions: MakeStylesDefinition<
     const overrides: MakeStylesMatchedDefinitions = {};
     let overridesCx: string = '';
 
-    classNames.forEach(className => {
-      if (typeof className === 'string') {
-        if (className === '') {
-          return;
-        }
+    // arguments are parsed intentionally manually to avoid double loops as TS transforms rest via an additional loop
+    for (let i = 2; i < arguments.length; i++) {
+      // eslint-disable-next-line prefer-rest-params
+      const classNames = arguments[i];
+
+      if (typeof classNames === 'string' && classNames !== '') {
+        // .split() is an expensive call, it's faster to ensure that string contains any spaces before splitting
+        if (classNames.indexOf(' ') === -1) {
+          const definition = DEFINITION_LOOKUP_TABLE[classNames];
 
-        className.split(' ').forEach(cName => {
-          if (DEFINITION_LOOKUP_TABLE[cName] !== undefined) {
-            overrides[DEFINITION_LOOKUP_TABLE[cName][0]] = DEFINITION_LOOKUP_TABLE[cName][1];
-            overridesCx += cName;
+          if (definition !== undefined) {
+            overrides[definition[0]] = DEFINITION_LOOKUP_TABLE[classNames][1];
+            overridesCx += classNames;
           } else {
-            nonMakeClasses += cName + ' ';
+            nonMakeClasses += classNames + ' ';
           }
-        });
+        } else {
+          classNames.split(' ').forEach(className => {
+            const definition = DEFINITION_LOOKUP_TABLE[className];
+
+            if (definition !== undefined) {
+              overrides[definition[0]] = definition[1];
+              overridesCx += className;
+            } else {
+              nonMakeClasses += className + ' ';
+            }
+          });
+        }
       }
-    });
+    }
 
     let matchedIndexes = '';
+    const matchedDefinitions: MakeStylesMatchedDefinitions[] = [];
 
-    const matchedDefinitions = resolvedDefinitions.reduce<MakeStylesMatchedDefinitions[]>((acc, definition, i) => {
-      const matcherFn = definition[0];
+    for (let i = 0, l = resolvedDefinitions.length; i < l; i++) {
+      const matcherFn = resolvedDefinitions[i][0];
 
       if (matcherFn === null || matcherFn(selectors)) {
-        acc.push(definition[2]);
+        matchedDefinitions.push(resolvedDefinitions[i][2]);
         matchedIndexes += i;
       }
-
-      return acc;
-    }, []);
+    }
 
     const overridesHash = overridesCx === '' ? '' : overridesCx;
     const cxCacheKey = options.renderer.id + matchedIndexes + '' + overridesHash;
```

Comment:
I also noticed that string concat is not cheap:
![image](https://user-images.githubusercontent.com/1207059/102519925-f871df00-4047-11eb-8e37-ee05dad7d87f.png)

we should avoid where possible.